lib/web/ui/server.go

```diff
@@ -152,6 +152,46 @@ func MakeKubeClusters(clusters []types.KubeCluster) []KubeCluster {
 	return uiKubeClusters
 }
 
+func MakeLegacyKubeClusters(servers []types.Server) []KubeCluster {
+	kubeClusters := map[string]*types.KubernetesCluster{}
+
+	// Get unique kube clusters
+	for _, server := range servers {
+		// Process each kube cluster.
+		for _, cluster := range server.GetKubernetesClusters() {
+			kubeClusters[cluster.Name] = cluster
+		}
+	}
+
+	uiKubeClusters := make([]KubeCluster, 0, len(kubeClusters))
+	for _, cluster := range kubeClusters {
```

Comment:
Suggestion: Instead of a 2-step iteration, you could do it all in the loop above. Swap `kubeClusters` for `seenClusters` (`map[string]struct{}`) and skip names you've seen before.